cli/server/server.go

```diff
@@ -285,13 +300,38 @@ func restoreDB(ctx *cli.Context) error {
 	if count == 0 {
 		count = lastBlock - start
 	}
+
+	var rootStart, rootSize uint32
+	if rootsIn != nil {
+		rootStart = rootsIn.ReadU32LE()
+		rootSize = rootsIn.ReadU32LE()
+		if rootsIn.Err != nil {
+			return cli.NewExitError(fmt.Errorf("error while reading roots file: %w", rootsIn.Err), 1)
+		}
+		if start < rootStart {
+			return cli.NewExitError(fmt.Errorf("roots file start from %d root, can't import %d", rootStart, start), 1)
+		}
+		lastRoot := rootStart + rootSize
+		if rootStart+count > lastRoot {
```

Comment:
I'm not sure it's correct treating it this way. State root chain is almost always one block behind and for a dump of blocks up to N height it's normal to only have dump of states up to N-1 height. This should be accounted for.